docs/PLUGIN_DIAGNOSTICS.md

```diff
@@ -0,0 +1,114 @@
+        Namespace: /intel/psutil/cpu/cpu5/softirq  Type: float64     Value: 0.69 
+        Namespace: /intel/psutil/cpu/cpu6/softirq  Type: float64     Value: 0.54 
+        Namespace: /intel/psutil/cpu/cpu7/softirq  Type: float64     Value: 0.31 
+        Namespace: /intel/psutil/cpu/cpu-total/softirq  Type: float64     Value: 50.52 
+        Namespace: /intel/psutil/cpu/cpu0/steal    Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu1/steal    Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu2/steal    Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu3/steal    Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu4/steal    Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu5/steal    Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu6/steal    Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu7/steal    Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu-total/steal  Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu0/guest_nice  Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu1/guest_nice  Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu2/guest_nice  Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu3/guest_nice  Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu4/guest_nice  Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu5/guest_nice  Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu6/guest_nice  Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu7/guest_nice  Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu-total/guest_nice  Type: float64     Value: 0 
+        Namespace: /intel/psutil/cpu/cpu0/user     Type: float64     Value: 2650.97 
+        Namespace: /intel/psutil/cpu/cpu1/user     Type: float64     Value: 2631.81 
+        Namespace: /intel/psutil/cpu/cpu2/user     Type: float64     Value: 2523.45 
+        Namespace: /intel/psutil/cpu/cpu3/user     Type: float64     Value: 2500.99 
+        Namespace: /intel/psutil/cpu/cpu4/user     Type: float64     Value: 2039.26 
+        Namespace: /intel/psutil/cpu/cpu5/user     Type: float64     Value: 2082.44 
+        Namespace: /intel/psutil/cpu/cpu6/user     Type: float64     Value: 2054.43 
+        Namespace: /intel/psutil/cpu/cpu7/user     Type: float64     Value: 1901.43 
+        Namespace: /intel/psutil/cpu/cpu-total/user  Type: float64     Value: 18384.81
+       [...]
+    ```
+
+- Times required to make operations, e.g.:
+    ```
+    [...]
+    printRuntimeDetails took 11.878µs
+    [...]
+    printConfigPolicy took 67.654µs 
+    [...]
+    printMetricTypes took 504.483µs 
+    [...]
+    printCollectMetrics took 7.470091ms 
+    [...]
+    showDiagnostics took 8.076025ms
+    ```
+## Note
+
+For now plugin diagnostics works only for collector plugins.
```

Comment:
Ok, now I find it.

Reply:
@kjlyon Great ideas! It is important that plugin must be written using one of our libs and diagnostics works also for streaming collectors.